docs/blueprints/EP025.rst

```diff
@@ -0,0 +1,111 @@
+
+
+Specification
+=============
+
+It is desirable to support ignored loops configuration that can be changed dynamically via API. Since this feature is relatively small and wouldn't require much extra entries or states, then using `topology` switches' and interfaces' metadata storage would simplify the implementation and maintenance since currently adding support to a new storage with storehouse requires a significant effort development-wise. In the future, this can be migrated to another storage (collection) when an alternative is available. The following metadata would be used:
+
+- ``switches.<dpid>.metadata.ignored_loops``:
+
+This is for configuring which loops should be ignored on a ``<dpid>``, the value is a list of list pair of interface port numbers. For instance, if it's desirable to ignore a loop between interface number 5 and 6, and also between port 1 and itself: ``[[5, 6], [1,1]]``
+
+- ``interfaces.<interface_id>.metadata.looped``:
+
+This ``looped`` attribute is for state management, it will contain the following object whenever a loop is in place, the ``detected_at`` string datetime will be in UTC without any ``Z`` modifier:
+
+.. code-block:: python3
+
+   {
+     'port_numbers': [<int>, <int>]
+     'detected_at': <str>, in this format %Y-%m-%dT%H:%M:%S
+   }
+
+Looped states should be removed if the interface has been disabled or deactivated, also there should be a periodic routine to remove the ``looped`` metadata interface state if the loop hasn't been detected anymore, this clean up routine should run at a lower pace than of_lldp polling, usually 3-4 times slower, and if the loop hasn't been seen within this time frame again then it would assume that the loop has stopped.
+
+The following options would be configurable on settings:
+
+- ``LLDP_LOOP_ACTIONS``:
+
+This is for configuring which actions should be taken once a loop is found. The actions are a list of strings, ``log`` and ``disable`` are both supported. The ``log`` action will log an warning message informing that a loop was detected on a dpid and on which interfaces. The ``disable`` action will disable an interface via the ``topology`` NApp (through the ``topology/v3/interfaces/<intf_id>/disable`` endpoint), and eventually enable it again if the loop stops (through the ``topology/v3/interfaces/<intf_id>/enable`` endpoint). For instance, ``['log', 'disable']`` would apply both actions to any loop that isn't ignored and is found. The default actions would be ``['log']`` based on what's expected to be used in production in this first iteration.
```

Comment:
Those are the "default" actions, right? Have you considered the possibility of having port-specific actions + default actions (which applies if there is no port-specific ones) ?

Reply:
Yes, they're default actions for any interface. Nope, I haven't considered in this iteration, I think it's a great idea, let me know if you think it's going to be used in prod @italovalcy, if it's likely then it might be worth adding in the scope of this iteration here. 

To support that isn't that far away, we've got a [method that is called whenever a non-ignored loop is found](https://github.com/kytos-ng/of_lldp/blob/feature/loop_detection/loop_manager.py#L99-L104), and it takes both interfaces, so it'd be a matter of parametrizing and passing the options o be checked if it has interface specific config. If we we'll try to support that in this iteration, it's also good to know if operators would need to change this settings dynamically via metadata/API or if on settings would be OK for now.

Reply:
I've updated on this commit https://github.com/kytos-ng/kytos/pull/181/commits/78210212713ca9033c2f56f8a00887f674105721 highlighting they're default actions.

Reply:
Thanks Vinicius! Well, I think it is not likely to be consumed right away, so I agree that we can do this next interaction.